### PR TITLE
move test to separate folders, add test for notifications

### PR DESCRIPTION
**Pull request by vasylkushnir**

### CODE REVIEW

File: src/services/notifications/notifications.test.ts [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+let notificationId: number;
+const notExistedNotificationId = '66c9a620-3bdc-11ec-a993-ff3e9e1971e7';
+const notificationInfo = {
+  description: 'test description',
+  userId,
+  type: 'REPORT_CREATED',
+};
+
+describe('\'notifications\' service', () => {
+  it('registered the service', () => {
```
Comment (m-ilchuk) [inline src/services/notifications/notifications.test.ts]: remove this

File: src/services/notifications/notifications.test.ts [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    expect(service).to.exist;
+  });
+  describe('create notification', () => {
+    before(async () => {
+      const { id } = await app.service('users').create(userInfo);
```
Comment (m-ilchuk) [inline src/services/notifications/notifications.test.ts]: In order to simplify this, you could create one user in global describe `before` and reuse it in every notification tests and remove  in `after`.

File: src/services/notifications/notifications.test.ts [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+const service = app.service('notifications');
+const userInfo = {
+  firstName: 'test user name',
+  lastName: 'test user last',
+  email: 'create_notification@example.com',
+  password: 'supersecret',
+};
+let userId!: number;
+let notificationId: number;
```
Comment (m-ilchuk) [inline src/services/notifications/notifications.test.ts]: number ?

File: src/services/notifications/notifications.test.ts [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+      notificationInfo.userId = id;
+    });
+
+    after(async () => {
+      await service.remove(notificationId);
+      await app.service('users').remove(userId);
+    });
+
+    it('should succeed - create notification', async () => {
+      const notificationResponse = await service.create(notificationInfo);
```
Comment (m-ilchuk) [inline src/services/notifications/notifications.test.ts]: why not to `const { description, id }  = await ... `  or just keep it as is and `expect(notificationResponse.id)`

also , check each param value or type

File: src/services/notifications/notifications.test.ts [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+      expect(notificationResponse).to.have.keys(['id', 'isRead', 'type', 'description', 'createdAt', 'updatedAt', 'userId']);
+    });
+
+    it('should fail - not existed userId', async () => {
+      const notificationInvalidUserId = {
+        description: 'test notification',
+        userId: '783deee0-3732-11ec-9301-13adbda06b66',
+      };
+      try {
+        await service.create(notificationInvalidUserId);
```
Comment (m-ilchuk) [inline src/services/notifications/notifications.test.ts]: put the data directly here, it's not needed to be in a separate variable

File: src/services/notifications/notifications.test.ts [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+    });
+
+    it('should fail - invalid data', async () => {
+      const notificationInvalidData = {
+        description: 12,
+        isRead: 'yes',
+        type: 'READ_WRITE',
+      };
+      try {
+        await service.create(notificationInvalidData);
```
Comment (m-ilchuk) [inline src/services/notifications/notifications.test.ts]: same here

File: src/services/notifications/notifications.test.ts [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+        expect.fail('call should have failed');
+      } catch (err: any) {
+        const { code } = err;
+        expect(code).to.equal(400);
```
Comment (m-ilchuk) [inline src/services/notifications/notifications.test.ts]: check error messages if possible

File: src/services/notifications/notifications.test.ts [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+        await service.get('7777');
+        expect.fail('call should have failed');
+      } catch (err: any) {
+        const { code } = err;
+        expect(code).to.equal(400);
+      }
+    });
+  });
+
+  describe('find notification', () => {
```
Comment (m-ilchuk) [inline src/services/notifications/notifications.test.ts]: add test, where you use pagination, check sorting

File: src/services/notifications/notifications.test.ts [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+
+    it('should succeed - get notification by id', async () => {
+      const response: any = await service.find();
+      const { total } = response;
+      expect(total).not.to.equal(0);
+      expect(response).to.have.keys(['total', 'limit', 'skip', 'data']);
+    });
+  });
+
+  describe('update (patch) notification status', () => {
```
Comment (m-ilchuk) [inline src/services/notifications/notifications.test.ts]: add test readMany notifications

**Review by m-ilchuk — CHANGES_REQUESTED** [review]

File: src/services/notifications/notifications.test.ts (hunk 1 above)
Comment (vasylkushnir) [reply src/services/notifications/notifications.test.ts]: moved to Before 9978e7fc0bb98929f17ca5c20384f8b3d56f0792

File: src/services/notifications/notifications.test.ts (hunk 2 above)
Comment (vasylkushnir) [reply src/services/notifications/notifications.test.ts]: fixed 9978e7fc0bb98929f17ca5c20384f8b3d56f0792

File: src/services/notifications/notifications.test.ts (hunk 3 above)
Comment (vasylkushnir) [reply src/services/notifications/notifications.test.ts]: changed to Id 9978e7fc0bb98929f17ca5c20384f8b3d56f0792

File: src/services/notifications/notifications.test.ts (hunk 4 above)
Comment (vasylkushnir) [reply src/services/notifications/notifications.test.ts]: fixed 9978e7fc0bb98929f17ca5c20384f8b3d56f0792

File: src/services/notifications/notifications.test.ts (hunk 5 above)
Comment (vasylkushnir) [reply src/services/notifications/notifications.test.ts]: fixed 9978e7fc0bb98929f17ca5c20384f8b3d56f0792

File: src/services/notifications/notifications.test.ts (hunk 6 above)
Comment (vasylkushnir) [reply src/services/notifications/notifications.test.ts]: fixed 9978e7fc0bb98929f17ca5c20384f8b3d56f0792

File: src/services/notifications/notifications.test.ts (hunk 7 above)
Comment (vasylkushnir) [reply src/services/notifications/notifications.test.ts]: add check for error message 9978e7fc0bb98929f17ca5c20384f8b3d56f0792

File: src/services/notifications/notifications.test.ts (hunk 8 above)
Comment (vasylkushnir) [reply src/services/notifications/notifications.test.ts]: tests added 9978e7fc0bb98929f17ca5c20384f8b3d56f0792

File: src/services/notifications/notifications.test.ts (hunk 9 above)
Comment (vasylkushnir) [reply src/services/notifications/notifications.test.ts]: added 9978e7fc0bb98929f17ca5c20384f8b3d56f0792

**Pull request merged**